app/lib/actions/katello/repository/destroy.rb

```diff
@@ -56,19 +56,21 @@ def finalize
 
         def handle_alternate_content_sources(repository)
           product = repository.product
-          content_type = repository.content_type
           repository.smart_proxy_alternate_content_sources.each do |smart_proxy_acs|
             plan_action(Pulp3::Orchestration::AlternateContentSource::Delete, smart_proxy_acs)
           end
 
-          # Remove the product from the ACS if it's empty.
-          # An ACS with only an empty product will not function correctly
-          ## because there will be no smart_proxy_alternate_content_sources.
-          if product.repositories.with_type(content_type).count == 1
+          # If there are no URLs for any of the remaining repos, remove the
+          # product from the ACS. An ACS with only an empty product will not
+          # function correctly because there will be no
+          # smart_proxy_alternate_content_sources.
+          contains_url = product.repositories.any? do |url_test_repo|
```

Comment:
We still want to limit this search to repositories of the same content_type as the ACS right?